packages/autorest.python/autorest/codegen/serializers/parameter_serializer.py

```diff
@@ -96,14 +96,65 @@ def serialize_path(
             [
                 '    "{}": {},'.format(
                     path_parameter.wire_name,
-                    self.serialize_parameter(path_parameter, serializer_name),
+                    ParameterSerializer.serialize_parameter(
+                        path_parameter, serializer_name
+                    ),
                 )
                 for path_parameter in parameters
             ]
         )
         retval.append("}")
         return retval
 
+    @staticmethod
+    def serialize_query_header(
+        param: Parameter, kwarg_name: str, serializer_name: str
+    ) -> List[str]:
+        if (
+            param.location == ParameterLocation.HEADER
+            and param.wire_name.lower() == "repeatability-request-id"
+        ):
+            return [
+                """if "Repeatability-Request-ID" not in _headers:""",
+                """    _headers["Repeatability-Request-ID"] = str(uuid.uuid4())""",
+            ]
+        if (
+            param.location == ParameterLocation.HEADER
+            and param.wire_name.lower() == "repeatability-first-sent"
+        ):
+            return [
+                """if "Repeatability-First-Sent" not in _headers:""",
```

Comment:
nit: you can take out this special logic for `Repeatability-First-Sent` and `Repeatability-Request-Id` etc and make a helper method